### DISCUSSION TITLE
Poor DatePicker UX

### DISCUSSION THREAD

**Discussion by megafinz** [keystonejs/keystone#7904] [Feature requests]
Suppose I have a `calendarDay` or `timestamp` field and I'm adding content in the Admin UI. It seems that currently the only way to enter the date is using the Calendar widget (I can't edit the date directly in the input). Unfortunately the widget doesn't provide year selection, so to enter the past date I have to click through all the months up until the desired one. It's not that bad for the dates that span a couple of recent years, but if I have to mark that something happened like 20 years ago I have to click the month button 200+ times which is problematic.

Am I missing something obvious here? I can't see any relevant settings in the `ui` section of the field setup.

**Comment by linus-ha** [comment]
I totally agree. Even the default HTML form field does this way better in any browser.

**Comment by dcousens** [comment]
This could be a bug report, or feature request - could work either way.
I'll change to `Feature request` for now, as there is no suitable "correct answer" (except if we fix the problem)

In any event, thanks for the feedback @megafinz and @linus-ha; this isn't being ignored.

**Comment by rankun203** [comment]
Also the `<` and `>` is shifting places, when we need to jump over years and do speed click, it's very easy to click in the wrong place and dismiss the dialog, then we have to start again.

<img width=506 src="https://user-images.githubusercontent.com/2988555/210298009-a0861ff2-1a5f-4956-9f6a-e41c60e4c3b9.gif" />

**Comment by tomcardoso** [comment]
+1'ing this. I will probably just write a custom field that uses the standard browser date picker instead in the meantime…

**Comment by BastianGanze** [comment]
Anyone looking for a custom component that just replaces the current one with the default html input can take the one I have written here:

https://github.com/BastianGanze/portfolio/blob/master/bg-cms/admin/components/HtmlDatePicker.tsx

Here is an example on how I use it:

https://github.com/BastianGanze/portfolio/blob/be94d02befe7d950c445f470c8d08451e8c96335/bg-cms/schema.ts#L134
